### PR TITLE
[RFR ]Widgetastic conversion of Chargeback Form

### PR DESCRIPTION
**Pull request by nachandr**
{{pytest: -v -k 'chargeback'}}

### CODE REVIEW

File: cfme/intelligence/chargeback/__init__.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        tree = ManageIQTree()
+
+
+@navigator.register(Server)
+class IntelChargeback(CFMENavigateStep):
+    VIEW = IntelChargebackView
+    prerequisite = NavigateToSibling("LoggedIn")
+
+    def step(self):
+        self.view.navigation.select("Cloud Intel", "Chargeback")
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/__init__.py]: You should use self.prerequisite_view.

File: cfme/intelligence/chargeback/assignments.py [hunk 2]
```diff
@@ -0,0 +1,185 @@
+# -*- coding: utf-8 -*-
+import cfme.fixtures.pytest_selenium as sel
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/assignments.py]: Can we get rid of this import?

File: cfme/intelligence/chargeback/rates.py [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+
+
+class AddComputeChargebackView(RatesView):
+    title = Text('#explorer_title_text')
+
+    description = Input(id='description')
+    currency = Select(id='currency')
+
+    @ParametrizedView.nested
+    class fields(ParametrizedView):  # noqa
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/rates.py]: :heart: Parametrized view :)

File: cfme/intelligence/chargeback/rates.py [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+    pretty_attrs = ['description']
+
+
+@navigator.register(StorageRate, 'All')
+class StorageRateAll(CFMENavigateStep):
+    VIEW = StorageChargebackView
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def step(self):
+        self.prerequisite_view.navigation.select('Cloud Intel', 'Chargeback')
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/rates.py]: I believe this should be a separate step as well.

File: cfme/intelligence/chargeback/rates.py [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+        )
+
+
+@navigator.register(StorageRate, 'New')
+class StorageRateNew(CFMENavigateStep):
+    VIEW = AddStorageChargebackView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.view.configuration.item_select("Add a new Chargeback Rate")
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/rates.py]: Here should be the prerequisite_view

**Comment by dajoRH** [comment]
Lint report for commit be10b1e959ff5c79cd7ce83ed84bc4cc42893fda:

`cfme/intelligence/chargeback/rates.py`:
- :red_circle: Line 8:1: **F401** *'utils.log.logger' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

File: cfme/intelligence/chargeback/rates.py [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+@navigator.register(ComputeRate, 'Edit')
+class ComputeRateEdit(CFMENavigateStep):
+    VIEW = EditComputeChargebackView
+    prerequisite = NavigateToSibling('Details')
+
+    def step(self):
+        self.view.configuration.item_select("Edit this Chargeback Rate")
+
+
+class StorageRate(ComputeRate, Updateable, Pretty, Navigatable):
```
Comment (mfalesni) [inline cfme/intelligence/chargeback/rates.py]: You can inherit just the ComputeRate, all the other three are already inherited :)

**Comment by dajoRH** [comment]
Lint report for commit a21fb55ce58d98574e80caa7bcf6b65ebd3362d4:

`cfme/intelligence/chargeback/rates.py`:
- :red_circle: Line 8:1: **F401** *'utils.log.logger' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by dajoRH** [comment]
Lint report for commit d3bb5530efda071cf895d96f9ad1753ec6ba773f:

`cfme/intelligence/chargeback/rates.py`:
- :red_circle: Line 8:1: **F401** *'utils.log.logger' imported but unused*

Please, rectify these issues :smirk: .


**No commit flaws detected**.

*CFME QE Bot*

**Comment by nachandr** [comment]
That's correct.The web_ui object will be converted later.

**Pull request merged**